common/web/keyboard-processor/src/text/keyEvent.ts

```diff
@@ -156,7 +159,7 @@ export default class KeyEvent implements KeyEventSpec {
       // the actual keyname instead.
       mappingEvent.kName = 'K_xxxx';
       mappingEvent.Lmodifiers = (shifted ? 0x10 : 0);  // mnemonic lookups only exist for default & shift layers.
-      var mappedChar: string = BASE_DEFAULT_RULES.forAny(mappingEvent, true);
+      var mappedChar: string = defaultRules.forAny(mappingEvent, true);
```

Comment:
The reason that `defaultRules` is extendable is so that `app/browser` use can override certain keys to perform 'command' actions.  Namely, within certain contexts, `K_TAB` can tab among editable elements and `K_ENTER` can submit forms.

Here, we don't care about that.  We only care about the default text-code value corresponding to the key when doing mnemonic key remapping.  We don't need overrides from higher levels of the engine here - the base-level `DefaultRules` implementation will suffice.

Of course, the current problem is that the current definition of `DefaultRules` (being moved to `js-processor`) actually refers to the `RuleBehavior` type.  That said, please note the pattern of its use - it's _just_ to facilitate the tracking of warning and error messages.  We can establish a different pattern for that, returning messages along with the char in some sort of object.  Code that cares about the messages can then forward that into `RuleBehavior` within `js-processor`.

If modified in such a manner, `DefaultRules` would no longer refer directly to `RuleBehavior`; from there, we'd be safe to pull `DefaultRules` back into `keyboard-processor` / the new child project `keyboard`, where it _is_ needed due to mnemonic processing for keyboard layouts.  (We may need access to it when building OSKs, even without `js-processor` available, in the future when we start integrating the OSK within the desktop apps.)

Reply:
Code reference for why _this method_ (`setMnemonicCode`), the sole point left within the eventual `engine/keyboard` that relies on `defaultRules`, should still stay within it:

https://github.com/keymanapp/keyman/blob/bba694c306d17a841b2e86606de86638bbfa4397/common/web/keyboard-processor/src/keyboards/keyboard.ts#L694-L701

The OSK will need some strategy or proper format for the events that are signaled by OSK key presses, and this logic helps ensure that the correct one is produced and provided to the OSK's consumer... even if, in the future, that consumer isn't always the Web engine.

Keeping it here then necessitates keeping the basic `DefaultRules` definition within the same child project, rather than a permanently-Web-only child project.

Reply:
Admittedly, yeah, it'd be ideal if `DefaultRules` could live within `js-processor`, but to me, this function presents a strong enough case to override that preference.  We need it or a near-equivalent due to OSK event requirements.

Reply:
Done